src/odemis/driver/xt_client.py

```diff
@@ -1113,7 +1113,7 @@ def __init__(self, name, role, parent, rng=None, **kwargs):
         self._executor = CancellableThreadPoolExecutor(max_workers=1)  # one task at a time
 
         self.position = model.VigilantAttribute({}, unit=stage_info["unit"],
-                                                readonly=True)
+                                                readonly=True, getter=self._getPosition)
```

Comment:
Let's not do that. A getter is called *every* time the VA is read. That means it's going to be very costly (in time), and if for some reason the connection is down, it'll be blocking, really annoying.
That's why I'd recommend what we normally do: update during/after a move, and otherwise, from time to time. It's enough for most use-cases.